src/relayserver/RelayServer.js

```diff
@@ -574,13 +573,13 @@ class RelayServer extends EventEmitter {
    */
   async _resendUnconfirmedTransactions (blockHeader) {
     // repeat separately for each signer (manager, all workers)
-    let receipt;
-    [0, 1].forEach(signerIndex => {
-      const ret = this._resendUnconfirmedTransactionsForSigner(blockHeader, signerIndex)
+    let receipt
+    for (const signerIndex of [0, 1]) {
+      const ret = await this._resendUnconfirmedTransactionsForSigner(blockHeader, signerIndex)
```

Comment:
This bug is so hard to detect, when you return a promise and you await for `_resendUnconfirmedTransactions` when you call it. If you have only one unconfirmed tx then you won't ever notice it.